modules/scheduler-remote/src/main/java/org/opencastproject/scheduler/remote/SchedulerServiceRemoteImpl.java

```diff
@@ -961,6 +756,56 @@ public String getScheduleLastModified(String agentId) throws SchedulerException
     throw new SchedulerException("Unable to get recordings from remote scheduler service");
   }
 
+  @Override
+  public Opt<MediaPackage> getCurrentRecording(String captureAgentId) throws SchedulerException, UnauthorizedException {
+    HttpGet get = new HttpGet(UrlSupport.concat("currentRecording", captureAgentId));
+    HttpResponse response = getResponse(get, SC_OK, SC_NO_CONTENT, SC_UNAUTHORIZED);
+    try {
+      if (SC_OK == response.getStatusLine().getStatusCode()) {
+        String mediaPackageXml = EntityUtils.toString(response.getEntity(), UTF_8);
+        MediaPackage event = MediaPackageParser.getFromXml(mediaPackageXml);
+        logger.info("Successfully get current recording of agent {} from the remote scheduler service", captureAgentId);
+        return Opt.some(event);
+      } else if (SC_UNAUTHORIZED == response.getStatusLine().getStatusCode()) {
+        logger.info("Unauthorized to get current recording of agent {}", captureAgentId);
+        throw new UnauthorizedException("Unauthorized to get current recording of agent " + captureAgentId);
+      } else {
+        return Opt.none();
+      }
+    } catch (UnauthorizedException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new SchedulerException("Unable to get current recording from remote scheduler service: " + e);
```

Comment:
```suggestion
      throw new SchedulerException("Unable to get current recording from remote scheduler service: ", e);
```

Reply:
I will change all the places in SchedulerRemoteServiceImpl where the `SchedulerException("some message: " + e)` pattern is used to `SchedulerException("some message",  e)`. However, I would like to not go any further at this point since it is not the scope of this PR to completely rework Opencasts Exception Handling.

Reply:
Makes sense. Thanks.
I've also deliberately only complained about the newly introduced exceptions ;-P